app/models/user.rb

```diff
@@ -0,0 +1,33 @@
+class User < ApplicationRecord
+  devise :confirmable, :database_authenticatable, :registerable, :recoverable,
+    :rememberable, :trackable, :validatable
+
+  after_create :reset_confirmation_token
+
+  def attempt_set_password(params)
+    unless params[:password] == params[:password_confirmation]
```

Comment:
Need to be careful we're not comparing two empty strings here.

Reply:
Devise has a validation on emails being of a certain length so thats handled elsewhere.